content/docs/practices/naming.md

```diff
@@ -39,6 +39,26 @@ A metric name...
    (for a pseudo-metric that provides [metadata](https://www.robustperception.io/exposing-the-software-version-to-prometheus) about the running binary)
  * <code>data\_pipeline\_last\_record\_processed\_<b>timestamp_seconds</b></code>
   (for a timestamp that tracks the time of the latest record processed in a data processing pipeline)
+* ...that has a `unit` in its name, must always have that `unit` as the last word. Note that an accumulating count such as `total` will be the last word, in addition to the unit if applicable. Examples,
+  * <code>request\_size\_<b>bytes</b></code>
+  * <code>request\_size\_<b>bytes</b>\_<b>total</b></code>
+  * <code>process\_cpu\_<b>seconds</b></code>
+  * <code>process\_cpu\_<b>seconds</b>\_<b>total</b></code>
+* ...that has a accumulating count such as `total`, can have two different naming scenarios. 
+  * If the metric name has a unit, then the unit must be the last word before the accumulating count. Examples,
+    * <code>process\_cpu\_<b>seconds</b>\_<b>total</b></code>
+    * <code>request\_size\_<b>bytes</b>\_<b>total</b></code>
+  * If the metric name does'nt have a real unit, then the naming can follow any sorting order in a way that it fits your use case, such as `grouping similar metric names`. Examples,
```

Comment:
I'm not sure I agree with this section. The example `connections` is a real countable unit.

However, we have agreed in the past that unit modifiers are prefered to be before, but are still allowed to be after. For example, both `process_thing_failed_bytes_total` and `process_thing_bytes_failed_total` are both valid.

Reply:
I understand, i can try to rephrase that. 

But with regards to your overall comment, I understood that only standalone units which are measurable, like "seconds", "bytes" are considered to be a real unit. In a different context "connections" could be a unit, but it does'nt represent a unit of measurement though.

Correct me if i am wrong, ill read the discussion for this issue again to see if i can rephrase these lines.

Thanks for the feedback, will get back with this issue soon.

Reply:
I understood the previous discussions such that we had reached the consensus that we won't consider things like `connections` a "real" unit for the sake of metric naming. (I'm not sure, though, if we should use the wording "real unit" in the doc here.) If my understanding is flawed, maybe @juliusv and @SuperQ could go back to the drawing board and find out what the consensus is.

Very generally, this document is just a recommendation anyway, so we aren't defining a spec here and maybe don't have to be super precise.

On the other hand, OpenMetrics in its current state specifies about the UNIT metadata field:

> Unit specifies MetricFamily units. If non-empty, it MUST be a suffix of the MetricFamily name separated by an underscore. Be aware that further generation rules might make it an infix in the text format.

I'm not sure if we are ever going to enforce this in Prometheus (and the strict requirement might even be lifted in OM v2), but since the OM standard was evolved from existing best practices, it gives us some hints. I would believe the following to be true:
- A "real" unit is one that would show up in the UNIT metadata field. (E.g. we would probably not have a line like `# UNIT go_sql_stats_connections_idle connections` in the OM exposition.)
- Such a "real" unit really should be the last thing before any other suffix like `_total` (in OM, it's a MUST even).

So maybe that's what should go in here: Somehow explain what "real" units are (using a better wording than just saying "real unit"), and then have a very strong recommendation to put them last.